feat(broker): support custom logging configuration

npepinpe opened the pull request:
**Description**

Adds support for basic log level setting by mapping the value `logLevel` to the environment variable `ZEEBE_LOG_LEVEL`, and raises the log level to `WARN`. Also adds optional support to provide own log4j2.xml configuration file, such that the user can change to, say, JSON logging. This is done through an optional config map that is mounted to `/usr/local/zeebe/conf/log4j2.xml`.

Closes #2

salaboy commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **npepinpe**

If they are not already assigned, you can assign the PR to them by writing `/assign @npepinpe` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=zeebe-io%2Fzeebe-cluster-helm).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/zeebe-io/zeebe-cluster-helm/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["npepinpe"]} -->

npepinpe commented:
To set a custom `log4j2.xml`, a user could define the following value:

```yaml
log4j2: |
  <?xml version="1.0" encoding="UTF-8"?>
  <Configuration status="WARN" shutdownHook="disable">

    <Properties>
      <Property name="log.path">${sys:app.home}/logs</Property>
    </Properties>

    <Appenders>
      <Console name="Console" target="SYSTEM_OUT">
        <JsonLayout />
      </Console>

      <RollingFile name="RollingFile" fileName="${log.path}/zeebe.log"
        filePattern="${log.path}/zeebe-%d{yyyy-MM-dd}-%i.log.gz">
        <JsonLayout />
        <Policies>
          <TimeBasedTriggeringPolicy/>
          <SizeBasedTriggeringPolicy size="250 MB"/>
        </Policies>
      </RollingFile>
    </Appenders>

    <Loggers>
      <Logger name="io.zeebe" level="${env:ZEEBE_LOG_LEVEL:-info}"/>
      <Logger name="io.atomix" level="${env:ATOMIX_LOG_LEVEL:-warn}"/>

      <Root level="info">
        <AppenderRef ref="RollingFile"/>

        <!-- remove to disable console logging -->
        <AppenderRef ref="Console"/>
      </Root>
    </Loggers>

  </Configuration>
```

And this will be the file that is mounted and used by Zeebe when initially configuring log4j2. Should be sufficient for most use cases.

npepinpe commented:
/ok-to-test

Zelldon commented on zeebe-cluster/templates/configmap.yaml:
```diff
@@ -24,6 +24,10 @@ data:
     exec /usr/local/zeebe/bin/broker
+  {{- if .Values.log4j2 }}
+  log4j2.xml: |
+    {{ .Values.log4j2 | indent 4 | trim }}
```
Please add this to the read me

Zelldon commented on zeebe-cluster/values.yaml:
```diff
@@ -15,6 +15,7 @@ cpuThreadCount: "2"
 ioThreadCount: "2"
 pvcSize: "10Gi"
 pvcAccessModes: [ "ReadWriteOnce" ]
+logLevel: warn
```
why warn?

npepinpe replied:
I imagine if you're running this in production, you don't care about anything but warning/errors. Possibly you'd want INFO, depending on your level of paranoia. Is debug or info level logging interesting for people running Zeebe who are not Zeebe devs/contributors/testers?

npepinpe replied:
I set it down to INFO; debug should be mostly for us, don't see it as useful in production.

salaboy commented:
@npepinpe: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
serverless-jenkins | 2b5f66636e1aea5d567f53e428dc32e6e13d27ba | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

salaboy commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **npepinpe**
You can assign the PR to them by writing `/assign @npepinpe` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=zeebe-io%2Fzeebe-cluster-helm).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](zeebe-io/zeebe-cluster-helm/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["npepinpe"]} -->

salaboy commented:
@npepinpe: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | ea8056d0dd2c588cdb61b2e5a95c777dda4a1c36 | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

salaboy commented:
@npepinpe: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | 92056e24b0668331491a1c4f5168aa57be3c912d | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

npepinpe commented:
Updated to the latest version and also added it to the gateway; also lowered log level to info, it makes more sense than warn (though I still don't thing debug is a good one).

npepinpe commented:
I've tried both, and the file is correctly overwritten with the contents of my config map.

salaboy commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/zeebe-io/zeebe-cluster-helm/pull/40#pullrequestreview-442379991" title="Approved">Zelldon</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=zeebe-io%2Fzeebe-cluster-helm).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](zeebe-io/zeebe-cluster-helm/blob/master/OWNERS)~~ [Zelldon]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->
